### PR TITLE
fix: [CDS-87114]: filtering clusters from linkedcluster for environment

### PR DESCRIPTION
**Pull request by sahithikolichala**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

#### [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

Problem:
We were showing all the clusters , this would result into execution error
Solution:
Filtered With the linkedClusters
https://github.com/harness/harness-core-ui/assets/57372986/d94b6739-f0d4-4944-9a04-c890fc5d9ae9

### CODE REVIEW

**Review by cjlee01 — DISMISSED** [review]

**Comment by sahithikolichala** [comment]
trigger messagecheck

**Comment by sahithikolichala** [comment]
trigger messagecheck

**Review by cjlee01 — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/GitOpsCluster/GitOpsCluster.tsx [hunk 1]
```diff
@@ -123,6 +123,8 @@ const GitOpsCluster = (props: {
       accountIdentifier={accountId}
       projectIdentifier={projectIdentifier}
       orgIdentifier={orgIdentifier}
+      linkedClusters={data?.data?.content || []}
+      environmentIdentifier={props.envRef}
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/GitOpsCluster/GitOpsCluster.tsx:127]: I do not understand much around GitOps. But, don't you have use case of multi environments? Is `props.envRef` also handling multi environment use case?

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/GitOpsCluster/GitOpsCluster.tsx (hunk 1 above)
Comment (sahithikolichala) [reply src/modules/75-cd/components/EnvironmentsV2/EnvironmentDetails/GitOpsCluster/GitOpsCluster.tsx:127]: yes it does, when there are multiple environments
For each environment, there would be a cluster dropdown , so it picks and filters accordingly

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**